### DISCUSSION TITLE
Query regarding your editor platform's form creation and integration capabilities.

### DISCUSSION THREAD

**Discussion by rajputakku** [Sunbird-inQuiry/Community#75] [Q&A]
Hi @sajeshkayyath
I have a few queries that I would like to discuss with you regarding your editor platform's form creation and integration capabilities.
Currently, we have integrated [Jotform](https://www.jotform.com/) into our platform, which enables us to create forms in multiple languages (such as Hindi and Marathi) and add various question types, including single and multiple choice, image checkboxes, and audio files in the assessment forms. As we have already created several live forms on our platform, we are looking for a compatible solution to migrate our existing forms.

Thus, I would like to inquire whether your platform can provide similar form creation and integration facilities. To give you a better understanding of our requirements, I have attached a few reference forms from Jotforms [jotform example 1](https://form.jotform.com/220752237822453) [jotform example 2](https://form.jotform.com/223518731019453)

Additionally, I have a query about the mandatory fields (such as Board/Syllabus, Medium etc) in your question set. Are they fixed or dynamic? As we do not require all the information, it would be helpful to know whether we can customise the mandatory fields to only keep the required ones. If your platform allows for self-creation of forms with customizable mandatory fields, that would be ideal.

I look forward to hearing from you.

cc @krgauraw / @rajnishdargan / @vaibhavbhuva / @pallakartheekreddy / @PoojaJManjooran / @kameshbhr / @surendrasinghs

**Comment by sajeshkayyath** [comment]
@rajputakku - as discussed, we are checking this from our end and will let you know.

**Reply by sajeshkayyath** [reply]
@rajputakku - please find the response from our side.

_`I have attached a few reference forms from Jotforms [jotform example 1](https://form.jotform.com/220752237822453) [jotform example 2](https://form.jotform.com/223518731019453)`_

As per the samples given, we understand that this is equivalent to questions from a question bank that supports single and multiple choice, image checkboxes, and audio files. 

From inQuiry perspective we currently support single and multiple choice, image checkboxes. That audio support is not readily available right now, but inQuiry can be extended easily to support the same. The audio feature is supported by the API and the editor / player alone needs to be extended.

_`As we have already created several live forms on our platform, we are looking for a compatible solution to migrate our existing forms.`_
As per what I understand from this Jotforms [thread](https://www.jotform.com/answers/4700432-how-to-bulk-export-all-of-the-forms-we-created) is that,

> _You can download all of your forms as HTML files, submission data as a CSV file, and uploaded files as a single ZIP file using our Export Data feature._

You will have to evaluate how to export the questions from the downloaded data.

Currently there is no off the shelf migration utility from inQuiry to export and load the questions to question bank. You will have to build a custom tool for the same. How we are seeing it to be done is as per the below high level process once you have the questions exported.

1. Export the question from the [Jotform](https://www.jotform.com/).
2. Pass this as an input to the newly build custom tool
3. The custom tool then generating the questions in QuML format
4. This QuML compliant questions can then be passed on the inQuiry service to create the questions and question sets.


_`Thus, I would like to inquire whether your platform can provide similar form creation and integration facilities. `_
Similar form creation facility do exist in the inQuiry Editor. It is configurable forms which you can customise it according to your needs.

_`Additionally, I have a query about the mandatory fields (such as Board/Syllabus, Medium etc) in your question set. Are they fixed or dynamic?`_
The forms are configurable and the mandatory fields can be modified. They are dynamically determined based on the configuration returned by the API.

cc @krgauraw / @rajnishdargan / @vaibhavbhuva / @pallakartheekreddy / @PoojaJManjooran / @kameshbhr / @surendrasinghs
